beam/__init__.py

```diff
@@ -54,54 +58,269 @@ class SupportedUsageScope(Enum):
+        "probe",
+        help="Searches the registry for Extractors available for <input_type>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for Extractors
+            matching the provided input_type and returns the full definition of that
+            Extractor.
+        """,
+    )
 
-    args = argparser.parse_args()
+    probe.add_argument(
+        "input_type",
+        help="FileType.ID for which available Extractors are to be looked up.",
+    )
+    probe.set_defaults(func=fetch_registered_extractors)
+
+    yard = subparsers.add_parser(
+        "yard",
+        help="Searches the registry for an Extractors matching <extractor_id>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for an Extractor
+            that matches the provided extractor_id and returns its full definition.
+        """,
+    )
 
-    extract(
-        input_path=args.infile,
-        input_type=args.filetype,
-        output_path=args.outfile,
-        preferred_mode=SupportedExecutionMethod.CLI,
+    yard.add_argument(
+        "extractor_id",
+        help="Extractor.ID of the requested extractor.",
     )
+    yard.set_defaults(func=search_extractor)
+
+    install = subparsers.add_parser(
+        "install",
+        help="Installs the Extractor matching <extractor_id>.",
+        description=f"""
+            Searches the chosen registry (default: {REGISTRY_BASE_URL}) for an Extractor
+            that matches the provided extractor_id and installs it into the current
+            Python env.
+        """,
+    )
+
+    install.add_argument(
+        "extractor_id",
+        help="Extractor.ID of the requested extractor.",
+    )
+    install.add_argument(
+        "--no-use-venv",
+        action="store_false",
+        dest="use_venv",
+        help="""
+            Do not install into a separate venv. Note: this will try to install the
+            Extractor into your current Python environment, using pip!
+        """,
+        default=True,
+    )
+    install.set_defaults(func=install_extractor)
+
+    for p in probe, beam, yard, install:
+        p.add_argument(
+            "--verbose",
+            help="Set verbosity level to DEBUG.",
+            action="store_true",
+            default=False,
+        )
+
+        p.add_argument(
+            "--registry-base-url",
+            help="Base URL of the registry to use",
+            default=REGISTRY_BASE_URL,
+        )
+
+    args = parser.parse_args()
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+    delattr(args, "verbose")
+    logger.debug("loglevel set to DEBUG")  #
+
+    func = args.func
+    delattr(args, "func")
+    try:
+        ret = func(**vars(args))
+        if ret is not None and len(ret) > 0:
+            print(ret)
+    except RuntimeError as e:
+        print(e)
```

Comment:
I think for the CLI, we should try to return valid JSON rather than printing Python dictionaries so it can be piped into other tools, e.g.,

```suggestion
    try:
        ret = func(**vars(args))
        if ret is not None and len(ret) > 0:
            print(json.dumps(ret))
    except RuntimeError as e:
        print(e)
```

This might need some better error handling though.

Reply:
Yeah that's true, I forgot a printed python `dict` is not `json`.

Reply:
I've added a guarded JSON dump, also when using `probe` you now get e.g. `{"id": "biologic-mpr", "registered_extractors": ["yadg", "galvani"]}` back rather than just the list so its valid JSON.